### PR TITLE
Fix jupyter widget build

### PR DESCRIPTION
**Pull request by tafsiri**
Fix the JavaScript build step for the Jupyter widget.

- Add some files from @mkery that were missing in the initial PR. They were skipped because of the .gitignore which has been updated as well.
- Update NeoWidget.js to be compatible with ipywidgets 8.x (https://ipywidgets.readthedocs.io/en/latest/migration_guides.html#backbone-extend)

### CODE REVIEW

**Comment by tafsiri** [comment]
@mkery this is a PR into your earlier PR. I can confirm that the README instructions works for me in a fresh python env.

File: py-notebook/jupyter-widget/js/lib/NeoWidget.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
+var widgets = require("@jupyter-widgets/base");
+var _ = require("lodash");
```
Comment (domoritz) [inline py-notebook/jupyter-widget/js/lib/NeoWidget.js:2]: Where does this get used?

**Review by domoritz — APPROVED** [review]
Looks great. Do we want to include `py-notebook/jupyter-widget/js/lib/confMat.js` in the repo or just build it on the fly when we make a release? I would prefer the latter.

File: py-notebook/jupyter-widget/js/lib/NeoWidget.js (hunk 1 above)
Comment (tafsiri) [reply py-notebook/jupyter-widget/js/lib/NeoWidget.js:2]: Nowhere so far, but it may be related to the TODO tasks still pending?

**Comment by tafsiri** [comment]
@domoritz re `py-notebook/jupyter-widget/js/lib/confMat.js` I would recommend checking it in. When working on a a new release a build step should build it as needed. But checking it in will make it easier to reproduce the exact state of different releases with minimal effort (build tool output can drift), given the current level of development of this project I think that would be simpler.

**Comment by domoritz** [comment]
Okay, sounds good.

**Pull request merged**